scipy/optimize/tests/test_least_squares.py

```diff
@@ -430,6 +430,27 @@ def test_bounds_shape(self):
                 method=self.method)
             assert_allclose(res.x, [0.0, 0.5], atol=1e-5)
 
+    def test_bound_instances(self):
+        res = least_squares(fun_trivial, 0.5, bounds=Bounds())
+        assert_allclose(res.x, 0.0, atol=1e-4)
+
+        res = least_squares(fun_trivial, 3.0, bounds=Bounds(lb=1.0))
+        assert_allclose(res.x, 1.0, atol=1e-4)
+
+        res = least_squares(fun_trivial, 0.5, bounds=Bounds(lb=-1.0, ub=1.0))
+        assert_allclose(res.x, 0.0, atol=1e-4)
+
+        res = least_squares(fun_trivial, -3.0, bounds=Bounds(ub=-1.0))
+        assert_allclose(res.x, -1.0, atol=1e-4)
+
+        res = least_squares(fun_2d_trivial, [0.5, 0.5],
+                            bounds=Bounds(lb=[-1.0, -1.0], ub=1.0))
+        assert_allclose(res.x, [0.0, 0.0], atol=1e-5)
+
+        res = least_squares(fun_2d_trivial, [0.5, 0.5],
+                            bounds=Bounds(lb=[0.1, 0.1]))
+        assert_allclose(res.x, [0.1, 0.1], atol=1e-5)
+
```

Comment:
Could you please add tests to solve same optimization problems in  `test_bounds_shape` using `Bounds` instance here?
 Especially, we should check it works or not in ragged bounds case like `bounds=([0.3, 0.2], 3.0)` and using `jac` or not.